dll/win32/shell32/shlexec.cpp

```diff
@@ -2066,21 +2066,14 @@ static BOOL SHELL_execute(LPSHELLEXECUTEINFOW sei, SHELL_ExecuteW32 execfunc)
         TRACE("-- idlist=%p (%s)\n", sei_tmp.lpIDList, debugstr_w(wszApplicationName));
     }
 
-    if (sei_tmp.fMask & SEE_MASK_DOENVSUBST)
+    if ((sei_tmp.fMask & SEE_MASK_DOENVSUBST) && sei_tmp.lpFile && sei_tmp.lpFile[0])
     {
         WCHAR *tmp = expand_environment(sei_tmp.lpFile);
         if (tmp)
         {
             wszApplicationName.Attach(tmp);
             sei_tmp.lpFile = wszApplicationName;
         }
-
-        tmp = expand_environment(sei_tmp.lpDirectory);
```

Comment:
Are you sure that the directory shouldn't be expanded as well? Do we have tests to prove that?

Reply:
The above code (the working directory code) will expand instead of here.

Reply:
Ah, good catch.